velox/functions/sparksql/aggregates/SumAggregate.cpp

```diff
@@ -24,9 +25,33 @@ namespace facebook::velox::functions::aggregate::sparksql {
 namespace {
 template <typename TInput, typename TAccumulator, typename ResultType>
 using SumAggregate = SumAggregateBase<TInput, TAccumulator, ResultType, true>;
+
+TypePtr getDecimalSumType(
+    const TypePtr& resultType,
+    core::AggregationNode::Step step) {
+  if (exec::isPartialOutput(step)) {
```

Comment:
nit: Do we need this and the check below for `step == core::AggregationNode::Step::kSingle`? It looks like there are only two possibilities with resultType. Either it is a RowType where we always return childAt(0), or it's a decimal type where we return itself. Is this right?

Reply:
Yes, you are right. We do not need to consider the step, we can simply distinguish based on whether the `resultType` is ROW.